### ISSUE TITLE
Random node in cluster times out or very slow response

### ISSUE THREAD

**Issue by Caesar305** [apache/couchdb#2758]
## Description

We have a 3 node cluster where occasionally one of the nodes will either fail to respond (times out) or there is a significant delay in requests (30 seconds or longer).

## Steps to Reproduce

Setup a 3 node cluster on Ubuntu 18.04, 64 bit. Create about 1400 different databases with about 100 documents in each one. After a few weeks, suddenly the issue reported occurs. 

The following error appears in the log files repeatedly for 24 hours until the server is rebooted.
`[error] 2020-04-03T21:21:46.406281Z couchdb@db05.xxxx.com <0.14411.4655> b64446c5b6 rexi_server: from: couchdb@db06.xxxx.com(<7831.23329.314>) mfa: fabric_rpc:reduce_view/4 error:function_clause [{couch_db,incref,[undefined],[{file,"src/couch_db.erl"},{line,185}]},{couch_server,open,2,[{file,"src/couch_server.erl"},{line,85}]},{fabric_rpc,reduce_view,5,[{file,"src/fabric_rpc.erl"},{line,162}]},{rexi_server,init_p,3,[{file,"src/rexi_server.erl"},{line,140}]}]`

If you do not reboot the server, after 24-48 hours the server will resume normal operations however, another node in the cluster will experience this behavior. **Also note: when the issue occurs some databases on the failing node will work without issues while others will have the issue being reported.**

Rebooting all the servers in sequence resolves the issue until it reoccurs. 

## Expected Behaviour

Nodes should never have this issue according to my understanding.

## Your Environment

3 Nodes, running Ubuntu 18.04
16GB RAM, 16vCPUs, 1TB space (SSD drives)

curl -X GET http://127.0.0.1:5984
`{"couchdb":"Welcome","version":"2.3.1","git_sha":"c298091a4","uuid":"f5883ec656f70ecf62704f07e652b70e","features":["pluggable-storage-engines","scheduler"],"vendor":{"name":"The Apache Software Foundation"}}`

**Comment by Caesar305** [comment]
@wohali Any ideas on this issue? I noticed you created a bug for a similar issue about two years ago. 
#1356
#https://github.com/apache/couchdb/issues/1356

**Comment by wohali** [comment]
@Caesar305 In that case, the issue was a heavily overloaded machine. It was worked around by adding more CPU cores.

Have you tried CouchDB 3.0.0?

**Comment by Caesar305** [comment]
@wohali Yes, I just upgraded to 3.0 a few days ago and within 24 hours one of the nodes had the couchdb process take all the available memory (over 16GB) until it eventually crashed. I restarted the node that had this issue and it has been working fine for the last 48 hours.

**Comment by nicknaychov** [comment]
Hi 
we are also having same error using 3 node cluster running 3.1.1 . 
I noticed that I can replicate the error using the web interface and forcing it to open 100 DBs on a page. Web will start 100 DB GET's on page browsing. We have over 1000DBs.
 Other interesting fact is that after restart can not replicate it , but after couple days of processing requests. Here is our error:
`[error] 2021-07-09T09:26:09.692989Z couchdb@pbx1-z1.xxxxxx.ca <0.23888.629> 8a92a1d7df rexi_server: from: couchdb@xxxxxxx.ca(<0.21851.628>) mfa: fabric_rpc:reduce_view/4 error:function_clause [{couch_db,incref,[undefined],[{file,"src/couch_db.erl"},{line,187}]},{couch_server,open_int,2,[{file,"src/couch_server.erl"},{line,94}]},{couch_server,open,2,[{file,"src/couch_server.erl"},{line,84}]},{mem3_util,get_or_create_db,2,[{file,"src/mem3_util.erl"},{line,512}]},{fabric_rpc,reduce_view,5,[{file,"src/fabric_rpc.erl"},{line,160}]},{rexi_server,init_p,3,[{file,"src/rexi_server.erl"},{line,138}]}]`

Did anybody manage to solve it?
Thanks

**Comment by Andronikus** [comment]
Hi everyone,

It looks like this issue is open already long time ago... I have been experience a similar issue with Couch `3.2.2` too. 
Does anyone have any idea how to avoid or overcome it?

@nicknaychov

**Comment by nickva** [comment]
Closing the issue as `undefined` incref was fixed as mentioned above.
